Meadowlark-js/packages/meadowlark-core/src/plugin/listener/Publish.ts

```diff
@@ -0,0 +1,64 @@
+import Emittery from 'emittery';
+import { DeleteRequest } from '../../message/DeleteRequest';
+import { DeleteResult } from '../../message/DeleteResult';
+import { GetRequest } from '../../message/GetRequest';
+import { GetResult } from '../../message/GetResult';
+import { QueryRequest } from '../../message/QueryRequest';
+import { QueryResult } from '../../message/QueryResult';
+import { UpdateRequest } from '../../message/UpdateRequest';
+import { UpdateResult } from '../../message/UpdateResult';
+import { UpsertRequest } from '../../message/UpsertRequest';
+import { UpsertResult } from '../../message/UpsertResult';
+
+export const BEFORE_UPSERT_DOCUMENT = Symbol('BEFORE_UPSERT_DOCUMENT');
+export const AFTER_UPSERT_DOCUMENT = Symbol('AFTER_UPSERT_DOCUMENT');
+export const BEFORE_GET_DOCUMENT_BY_ID = Symbol('BEFORE_GET_DOCUMENT_BY_ID');
+export const AFTER_GET_DOCUMENT_BY_ID = Symbol('AFTER_GET_DOCUMENT_BY_ID');
+export const BEFORE_UPDATE_DOCUMENT_BY_ID = Symbol('BEFORE_UPDATE_DOCUMENT_BY_ID');
+export const AFTER_UPDATE_DOCUMENT_BY_ID = Symbol('AFTER_UPDATE_DOCUMENT_BY_ID');
+export const BEFORE_DELETE_DOCUMENT_BY_ID = Symbol('BEFORE_DELETE_DOCUMENT_BY_ID');
+export const AFTER_DELETE_DOCUMENT_BY_ID = Symbol('AFTER_DELETE_DOCUMENT_BY_ID');
+export const BEFORE_QUERY_DOCUMENTS = Symbol('BEFORE_QUERY_DOCUMENTS');
+export const AFTER_QUERY_DOCUMENTS = Symbol('AFTER_QUERY_DOCUMENTS');
```

Comment:
Can listeners filter messages by symbol? For example, to only react to the "AFTER_*" symbols.

Reply:
Yes, but via the registration methods in Subscribe.ts. The symbols are basically Emittery "topics" that we aren't exposing directly.